status/statistics.py

```diff
@@ -242,14 +242,22 @@ def get(self):
                 self.cleaning,
             )
         for fa in self.nanopore_flowcell_aggregates:
-            # Use |= to merge the resulting dictionary with what's already 
-            # inside data[fa], | works as a union for dictionaries.
-            data[fa] |= get_stats_data(
-                self.application.nanopore_runs_db,
-                self.nanopore_flowcell_aggregates[fa][0],
-                self.nanopore_flowcell_aggregates[fa][1],
-                self.cleaning,
+            nanopore_stats = get_stats_data(
+                    self.application.nanopore_runs_db,
+                    self.nanopore_flowcell_aggregates[fa][0],
+                    self.nanopore_flowcell_aggregates[fa][1],
+                    self.cleaning,
             )
+
+            # Use |= to merge the resulting dictionary with what's already
+            # inside data[fa][key], | works as a union for dictionaries.
+            # Doesn't work recursively though, so we have to do it for the bottom level only
+            for key, value in nanopore_stats.items():
+                if key in data[fa]:
+                    data[fa][key] |= value
```

Comment:
I'm okay with this as long as the non-commutative(?) property of this union does not give problems. E.g given dicts `a` and `b`, `a | b != b | a`

Reply:
Yes, it's true that it is non-commutative. Colliding keys inside the two dictionaries `data[fa][key]` and `value` will take the _value_ of `value`... 😂 . The [PEP](https://peps.python.org/pep-0584/#specification) probably expresses this better:

``` If a key appears in both operands, the last-seen value (i.e. that from the right-hand operand) wins```

I the usecase I have in mind (sequencing throughput plot) I think this is perfectly fine, but of course that might not be the case for other use cases.

Since the code was written in a very general way before, I didn't want to dumb it down and write it specific instead... Do you think that would instead be good in this case?